cmd/setup-etcd-environment/run.go

```diff
@@ -129,53 +164,82 @@ func runRunCmd(cmd *cobra.Command, args []string) error {
 		out = f
 	}
 
-	if runOpts.bootstrapSRV {
-		exportEnv["DISCOVERY_SRV"] = runOpts.discoverySRV
-	} else {
-		exportEnv["NAME"] = etcdName
+	if err := writeEnvironmentFile(exportEnv, out, true); err != nil {
+		return err
 	}
 
-	// enable etcd to run using s390 and s390x. Because these are not officially supported upstream
-	// etcd requires population of environment variable ETCD_UNSUPPORTED_ARCH at runtime.
-	// https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/supported-platform.md
-	arch := runtime.GOARCH
-	if strings.HasPrefix(arch, "s390") {
-		exportEnv["UNSUPPORTED_ARCH"] = arch
+	noExportEnv := map[string]string{
+		"IPV4_ADDRESS":      setupEnv.etcdIP,
+		"WILDCARD_DNS_NAME": fmt.Sprintf("*.%s", setupEnv.opts.discoverySRV),
 	}
-	if err := writeEnvironmentFile(exportEnv, out, true); err != nil {
-		return err
+	if setupEnv.etcdDNS != "" {
+		noExportEnv["DNS_NAME"] = setupEnv.etcdDNS
 	}
+	return writeEnvironmentFile(noExportEnv, out, false)
+}
 
-	return writeEnvironmentFile(map[string]string{
-		"IPV4_ADDRESS":      ip,
-		"DNS_NAME":          dns,
-		"WILDCARD_DNS_NAME": fmt.Sprintf("*.%s", runOpts.discoverySRV),
-	}, out, false)
+func (s *SetupEnv) getExportEnv() (map[string]string, error) {
```

Comment:
dumb q but why Export?

Reply:
We export certain variables so that the ENV is available not only to the container but also to the binary involved by the container as well.

Reply:
i think the func name really wasn't clicking with my brain. makes sense!